src/lib.rs

```diff
@@ -4,30 +4,31 @@
 //! ```no_run
 //! # fn main() -> Result<(), Box<dyn std::error::Error>> {
 //! let file = std::fs::read("foo.tar")?;
-//! # fn parse(file: &[u8]) -> Result<(), Box<dyn std::error::Error + '_>> {
-//! let (_, entries) = tar_parser2::parse_tar(&file[..])?;
+//! let entries = tar_parser2::parse_tar(&mut &file[..]).map_err(|err| err.into_inner().unwrap())?;
```

Comment:
given this is a fairly significant rewrite that we're not upstreaming, maybe we should rename to tar_parser3? or to binstall-tar-parser?

Reply:
I think tar-parser3 is a good name, having binstall- prefix makes it looks like a project only binstall can use.